processor/resourcedetectionprocessor/README.md

```diff
@@ -240,6 +240,40 @@ processors:
     override: false
 ```
 
+### AWS Lambda
+
+Uses the AWS Lambda [runtime environment variables](https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html#configuration-envvars-runtime)
+to retrieve the following resource attributes:
+
+[Cloud semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/cloud.md)
+
+* `cloud.provider` (`"aws"`)
+* `cloud.platform` (`"aws_lambda"`)
+* `cloud.region` (`$AWS_REGION`)
+
+[Function as a Service semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/faas.md)
+and [AWS Lambda semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-lambda.md#resource-detector)
+
+* `faas.name` (`$AWS_LAMBDA_FUNCTION_NAME`)
+* `faas.version` (`$AWS_LAMBDA_FUNCTION_VERSION`)
+* `faas.instance` (`$AWS_LAMBDA_LOG_STREAM_NAME`)
+* `faas.max_memory` (`$AWS_LAMBDA_FUNCTION_MEMORY_SIZE`)
+
+[AWS Logs semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/cloud_provider/aws/logs.md)
+
+* `aws.log.group.names` (`$AWS_LAMBDA_LOG_GROUP_NAME`)
+* `aws.log.stream.names` (`$AWS_LAMBDA_LOG_STREAM_NAME`)
+
+Example:
+
+```yaml
+processors:
+  resourcedetection/lambda:
+    detectors: [env, lambda]
+    timeout: 2s
```

Comment:
2 seconds is a long time in the lambda context, should we recommend a shorter timeout?

Reply:
I think I'd rather leave it as `2s` since that's that the rest of the examples use, but I could be convinced otherwise. I don't have a good sense for what influences performance for this processor. Let me know if you have a strong opinion about this.

Just a note in case it's not clear. You mention "in the lambda context". Since the [opentelemetry-lambda](https://github.com/open-telemetry/opentelemetry-lambda/blob/main/docs/design_proposal.md) project runs as a lambda [extension](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-extensions-api.html), a user's function handler execution time should not be dependant the otel collector's execution time nor this processor's execution time.

Reply:
> I don't have a good sense for what influences performance for this processor; theoretically these changes just read some environment variables which should be very quick

This is it and should be very quick.  2s is an eternity in that context.  There should be no other detectors running in this context as there will be no EC2, ECS, EKS, or other metadata available.

> Just a note in case it's not clear. You mention "in the lambda context". Since the [opentelemetry-lambda](https://github.com/open-telemetry/opentelemetry-lambda/blob/main/docs/design_proposal.md) project runs as a lambda [extension](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-extensions-api.html), a user's function handler execution time should not be dependant the otel collector's execution time nor this processor's execution time.

The Lambda collector extension is designed to be a synchronous export path to ensure that all telemetry is exported before the handler and extension report to the runtime API that execution of the invocation is complete so that the runtime does not freeze the execution environment while the collector extension is still processing data.  This is detailed in the [design doc](https://github.com/open-telemetry/opentelemetry-lambda/blob/main/docs/design_proposal.md#21-lambda-freeze).

Reply:
Ah, that's my misunderstanding about the execution context. I've gone ahead and updated the example to use a 0.2s timeout.